apps/web/src/components/Solana/ConnectWalletModal.tsx

```diff
@@ -0,0 +1,169 @@
+"use client";
+
+import { Box, Button, Text } from "@dex-web/ui";
+import type { WalletName } from "@solana/wallet-adapter-base";
+import { WalletReadyState } from "@solana/wallet-adapter-base";
+import type { Wallet } from "@solana/wallet-adapter-react";
+import { useWallet } from "@solana/wallet-adapter-react";
+import type { FC, MouseEvent } from "react";
+import {
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { createPortal } from "react-dom";
+import { twMerge } from "tailwind-merge";
+import { useWalletModal } from "./useWalletModal";
+import { WalletListItem } from "./WalletListItem";
+
+export interface WalletModalProps {
+  className?: string;
+  container?: string;
+}
+
+export const WalletModal: FC<WalletModalProps> = ({ container = "body" }) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const {
+    wallets,
+    select,
+    wallet: selectedWallet,
+    connecting,
+    connected,
+  } = useWallet();
+  const { setVisible } = useWalletModal();
+  const [fadeIn, setFadeIn] = useState(false);
+  const [portal, setPortal] = useState<Element | null>(null);
+
+  const [installedWallets, notInstalledWallets] = useMemo(() => {
```

Comment:
This useMemo will change every time wallets changes, I think we should look into ways we can do this without! 

Is this an expensive calculation?